File: 1/SelectionSort.java
class SelctionSort {
	
	public static void selectionSort(int array[]) {
		int n = array.length;
		
		for(int i=0; i<n-1; i++) {
			int minIdx = i;
			for(int j=i+1; j<n; j++) {
				if(array[j] < array[minIdx]) {
					minIdx = j; 
				}
			}
			
			int t = array[i];
			array[i] = array[minIdx];
			array[minIdx] = t;
		}
	}

	public static void main(String args[]) {
		int array[] = new int[]{8,5,2,9,5,6,3};
	
		selectionSort(array);
		
		for(int i : array) {
			System.out.print(String.format("%d\t", i));
		}
		
		System.out.println();
	}
}

File: 1/js/src/SelectionSort.ts
function swap(i: number, j: number, list: number[]) {
	if (i==j) return;

	const temp = list[i];
	list[i] = list[j];
	list[j] = temp;
}

function selectionSort(list: number[]) {
	const n = list.length;

	for(let i=0; i<n-1; i++) {
		let minIdx = i;

		for(let j=i+1; j<n; j++) {
			if(list[j] < list[minIdx]) {
				minIdx = j;
			}
		}

		swap(minIdx, i, list);
	}
}

(function main() {
	const list = [7,3,5,1,8,4,3];
	selectionSort(list);
	console.log(list);
})();
